Cocoaheads School_Lesson2.playground/Contents.swift

```diff
@@ -1,27 +1,54 @@
-class Stack {
-    private var array = [Int]()
+class Stack<T> {
+    /**
+     An array of 'T' elements
+    */
+    private var array = [T]()
     
-    func push(element: Int) {
+    /**
+     Adds a new element  on the top of the stack.
+     
+     - Parameter element: The element to add on the top of the stack.
+     - Complexity: O(1)
+     */
+    
+    func push(element: T) {
         array.append(element)
     }
     
-    func pop() -> Int? {
+    /**
+     Removes and returns the top element of the stack.
+     
+     - Returns: The top element of the stack if the stack is not empty; otherwise, nil.
+     - Complexity: O(1)
+    */
+    
+    func pop() -> T? {
         guard !array.isEmpty else {
             return nil
         }
         
         return array.removeLast()
     }
     
-    func top() -> Int? {
+    /**
+     Returns the top element of the stack, but does not remove it.
+     
+     - Returns: The top element of the stack if the stack is not empty; otherwise, nil.
+     - Complexity: O(1)
+     */
+    
+    func top() -> T? {
         return array.last
     }
 }
 
-class StackStatistics: Stack {
-    private var minElement: Int?
-
-    override func push(element: Int) {
+class StackStatistics<Element: Numeric & Comparable>: Stack<Element> {
+    /**
+     The minimum element in the stack.
+     */
+    private var minElement: Element?
+    
+    override func push(element: Element) {
```

Comment:
```suggestion
    override func push(element: Element) {
```
I think it is necessary to add a description for this method

Reply:
done